src/metrics/metrics.cc

```diff
@@ -0,0 +1,105 @@
+#ifdef HAVE_PROMETHEUS
+#include <prometheus/detail/builder.h>
+#include <prometheus/family.h>
+#include <pthread.h>
+#include <unistd.h>
+#include <exception>
+#include <memory>
+#endif
+
+#include "metrics.h"
+#include "slogger/slogger.h"
+
+
+namespace metrics {
+
+
+// Setting up new metrics:
+// 1. Add a MetricCounter type for the metric you want in this file and the
+// header file
+// 2. Create a setup function, indicating the metric family and counters. See
+// setup_client_packets for an example.
+// 3. Call that function in prometheus_loop
+
+// NOLINTBEGIN(cppcoreguidelines-avoid-non-const-global-variables)
+
+// These variables are all thread safe for now (They are wrapper classes that
+// simply check if metrics are enabled and call a function which is thread safe.
+// Metrics are enabled/disabled at startup, so that checked variable doesn't
+// change at all during the lifetime of the program.
+
+// Received client packets counter
+MetricCounter rx_client_packets;
+// Sent client packets counter
+MetricCounter tx_client_packets;
+
+// This is used to indicate for the metrics thread to stop, it's managed by a
+// single function (metrics_destroy) and read by prometheus_loop
+std::atomic<bool> destroy = false;
+
+std::unique_ptr<std::thread> thread;
+
+// NOLINTEND(cppcoreguidelines-avoid-non-const-global-variables)
+
+void metrics_destroy() {
+	if (thread != nullptr) {
```

Comment:
Please consider, if necessary, to change the places in the code related to comparisons to `nullptr` for a constant that improve readability.

Reply:
Not sure how making nullptr a constant named variable should help? In my mind it would be an unnecessary abstraction that's more likely to cause confusion (cause then you need to check what the constant name value is).

Reply:
it will help very much...
because there reason you compare it to nullptr,
and it would be nice you express it explicitly to save time of readers ....
```suggestion
constexpr std::thread* invalidThread = nullptr;
	if (thread != invalidThread) {
```